cheddar/cheddar-persistence/src/main/java/com/clicktravel/cheddar/infrastructure/persistence/database/configuration/ItemConfiguration.java

```diff
@@ -87,13 +89,21 @@ private void registerPrimaryKey(final PrimaryKeyDefinition primaryKeyDefinition)
     public void registerIndexes(final Collection<IndexDefinition> indexDefinitions) {
         for (final IndexDefinition indexDefinition : indexDefinitions) {
             final String indexPropertyName = indexDefinition.propertyName();
-            final PropertyDescriptor propertyDescriptor = properties.get(indexPropertyName);
-            if (propertyDescriptor == null) {
-                throw new IllegalStateException("No property found '" + indexPropertyName + "' for item :" + itemClass);
-            }
-            final Class<?> propertyType = propertyDescriptor.getPropertyType();
+            final Class<?> propertyType = getPropertyType(indexPropertyName);
             indexDefinition.setPropertyType(propertyType);
-            this.indexDefinitions.put(indexPropertyName, indexDefinition);
+            final StringBuilder indexPropertyNameBuilder = new StringBuilder(indexPropertyName);
+
+            if (CompoundIndexDefinition.class.isAssignableFrom(indexDefinition.getClass())) {
+                final CompoundIndexDefinition compoundIndexDefinition = (CompoundIndexDefinition) indexDefinition;
```

Comment:
I am not sure, but seeing as this is duplicated logic perhaps a simple helper method to construct a compound name from two component parts may make things clearer? Something like
```java
private String compoundName(String propertyName, String supportingPropertyName) {
    return propertyName + "_" + supportingPropertyName;
}
```
or the `StringBuilder` equivalent. I leave it you your judgement.